src/bsoncxx/view_or_value.hpp

```diff
@@ -66,14 +66,14 @@ class view_or_value {
     /// @param value
     ///   A Value type.
     ///
-    BSONCXX_INLINE view_or_value(Value&& value) : _value{std::move(value)}, _view{_value->view()} {
```

Comment:
Before, all of the type pairs that we used this template with (bsoncxx::document, bsoncxx::array) had Value classes with a .view() method.  This isn't something that we require of View/Value pairings, it was just a coincidence.  std::string doesn't have a .view(), it has no knowledge of the stdx::string_view class.  We do require that View types be constructible from Value types, though, so instead of relying on .view() we can use that property instead to construct Views everywhere in this class.